SourceCode/GPS/IO/TrackFiles.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using AgOpenGPS.Core.Models;
+
+namespace AgOpenGPS.IO
+{
+    public static class TrackFiles
+    {
+        /// <summary>
+        /// Load tracks from TrackLines.txt. Throws on malformed content or missing header.
+        /// </summary>
+        public static List<CTrk> Load(string fieldDirectory)
+        {
+            if (string.IsNullOrWhiteSpace(fieldDirectory))
+                throw new ArgumentNullException(nameof(fieldDirectory));
+
+            var result = new List<CTrk>();
+            var path = Path.Combine(fieldDirectory, "TrackLines.txt");
+            if (!File.Exists(path)) return result;
+
+            using (var reader = new StreamReader(path))
+            {
+                // Require header
+                var header = reader.ReadLine();
+                if (header == null || !header.TrimStart().StartsWith("$", StringComparison.Ordinal))
+                    throw new InvalidDataException("TrackLines.txt missing $ header.");
+
+                while (true)
```

Comment:
I think having an explicit exit condition would be better:
```suggestion
                while (!reader.EndOfStream)
```